### DISCUSSION TITLE
registerUriHandler - extension ID during development

### DISCUSSION THREAD

**Discussion by shoaib30** [microsoft/vscode-discussions#95] [Extension Development QnA] [answered]
Hi Everyone,
I am using registerUriHandler for my OAuth2 login flow.

From [Docs](https://code.visualstudio.com/api/references/vscode-api#window.registerUriHandler) :
> For example, if the my.extension extension registers a uri handler, it will only be allowed to handle uris with the prefix product name://my.extension.

where my.extension is the extension ID. 

I am not able to set the extension ID, and since I am still developing the value fetched from vscode.extensions.all for my extension does not have an ID.

Would be grateful if someone can help me form the appropriate callback url for the URI Handler

**Comment by shoaib30** [comment]
using 
```
console.log(
    vscode.extensions.all.map((x) => x.id).filter((x) => x.includes("extensionName"))
  );
```

I got `undefined_publisher.extensionName` as the ID.

using `vscode://undefined_publisher.extensionName` routes correctly to the extension


Is there no way to pre-set the value of ID?

**Reply by r3m0t** [reply]
The id comes from `package.json` fields `publisher` and `name`. I haven't had issues using this (hard coded) even in development mode. I never checked the value in `extensions.all` though.

**Reply by shoaib30** [reply]
I was missing the `publisher` key in my `package.json`. I used the `yo` generator to get started, it doesn't set the published.

Thank you for the pointer!

**Comment by shoaib30** [comment, marked as answer]
As pointed out by @r3m0t the ID for the extension is created using `<publisher_name>.<extension_name>` and the values of these two are taken from `package.json` fields `publisher` and `name` 

Hence the URI for the extension would be `vscode://publisherName.ExtensionName`

Reference: https://code.visualstudio.com/api/references/extension-manifest
